pkg/controllers/management/k3supgrade/deployPlans.go

```diff
@@ -190,3 +195,28 @@ func (h *handler) modifyClusterCondition(cluster *v3.Cluster, masterPlan, worker
 	return h.clusterClient.Update(cluster)
 
 }
+
+func upgradingMessage(concurrency int, nodes []string) string {
```

Comment:
This code looks fine but I don't understand the need for it if the results are equivalent to `strings.Join(nodes, ", ")`. However I do see value in this function if applies a truncating criteria (e.g. number of applying nodes, or length of the output string) with a conditionally appended ellipses to indicate truncation. For a large `plan.spec.concurrency` values it seems a bit much to burden a condition message (or is it reason?) with so much text.

Additionally, this message is rendered as-is in the UI correct? Something to know about the `plan.status.applying` slice is that the values therein are sorted in lexical order and not the order in which they were added to the list (which is essentially randomized per `plan.status.latestHash`). This means that for moderate to large `plan.spec.concurrency` values one is likely to see a bit more hopping around visually as nodes come in and out rather than what one might expect (e.g. an approximate fifo-like slide from right to left).

Reply:
Something that I missed initially is that the passed `concurrency` is a truncating criteria but its value is always sourced from the `plan.spec.concurrency` which should rarely if ever be less than the length of the `plan.status.applying` by design.